### ISSUE TITLE
The Kafka engine is unable to update and obtain the latest Worker address in a timely manner

### ISSUE THREAD

**Issue by FourSpaces** [ByConity/ByConity#457]
## Bug Report

The following error comes from the table: **cnch_system.cnch_kafka_log**
```
Check consumer status failed for kafka_consume.ck_logs_box_10_1688711127542132_0 due to: Code: 5038, e.displayText() = DB::Exception: 112:[E112]Not connected to 10.80.94.30:8124 yet, server_id=792 [R1][E112]Not connected to 10.80.94.30:8124 yet, server_id=792 [R2][E112]Not connected to 10.80.94.30:8124 yet, server_id=792 [R3][E112]Not connected to 10.80.94.30:8124 yet, server_id=792 SQLSTATE: HY000 (version 21.8.7.1)

Code: 5038, e.displayText() = DB::Exception: 112:[E112]Not connected to 10.80.87.32:8124 yet, server_id=793 [R1][E112]Not connected to 10.80.87.32:8124 yet, server_id=793 [R2][E112]Not connected to 10.80.87.32:8124 yet, server_id=793 [R3][E112]Not connected to 10.80.87.32:8124 yet, server_id=793 SQLSTATE: HY000 (version 21.8.7.1)

```

![image](https://github.com/ByConity/ByConity/assets/13359726/62baea0b-f528-4072-a7a7-e5782dfbacb6)

### Briefly describe the bug

After adjusting the resource configuration of the worker, the Kafka engine was unable to update and obtain the latest worker address in a timely manner, and stopped consuming Kafka data.

I need to restart 'Server' to restore normal consumption data.

### The result you expected

The Kafka engine can automatically obtain the latest healthy worker list after the worker status changes. Without the need to restart 'Server'

### How to Reproduce
1. Deploy byconity on k8s.

2. Properly configure the Kafka engine and obtain data normally.

3. Change the CPU/memory configuration of the worker and update the bytery on k8s.

4. You will find that you are unable to consume Kafka's data normally, which can be found on CNCH_ System.cnch_ Kafka_ Got an error in the log table.

### Version

0.1.1
dockerhub url : docker pull hustnn/byconity:0.1.0-fix1

**Comment by dmthuc** [comment]
Hi @FourSpaces , it is expected that kafka unable to consume for a short period of time when the worker is restarted. May I ask if you see the worker is shown up in system.workers after restarted or kafka unable to consume for a long time?

**Comment by dmthuc** [comment]
And does the worker ip and port are the same after restarted?

**Comment by FourSpaces** [comment]
@dmthuc  Before restarting the server, errors will continue to be reported. After restarting, the worker IP and port are different. They are deployed on K8S and are different every time they are restarted.

**Comment by dmthuc** [comment]
> @dmthuc Before restarting the server, errors will continue to be reported. After restarting, the worker IP and port are different. They are deployed on K8S and are different every time they are restarted.

Hi @FourSpaces , but does the info is updated in system.workers. Can you help do it again and check if worker info is updated in system.workers?

**Comment by FourSpaces** [comment]
@dmthuc  system.workers updated

**Comment by hillbun** [comment]
We  meet this also,   please fix asap,    this is critical.  thanks

**Comment by dmthuc** [comment]
Hi @zeromem please help to take a look

**Comment by hillbun** [comment]
only  write Worker pods restart, and IP changed,  will  cause this issue.

**Comment by dmthuc** [comment]
It is supposed to fixed in https://github.com/ByConity/ByConity/pull/496/commits/d0a9a9c16a7fa825b2bb9e72ba9a9006abbbc627

**Comment by hillbun** [comment]
when will this release?

**Comment by kevinthfang** [comment]
this has been merged and will be released in 0.2.0
